src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -839,7 +839,7 @@ protected override void DecodeWellKnownAttributeImpl(ref DecodeWellKnownAttribut
 
         private bool IsValidUnscopedRefAttributeTarget()
         {
-            return ParameterHelpers.IsRefScopedByDefault(this);
+            return UseUpdatedEscapeRules && RefKind != RefKind.None;
```

Comment:
If we make the change to only condition the declared escape rules on LangVersion, we might want to split this into one error for applying to a by-value parameter and a separate LangVersion error when used in C# 10 or prior.

I imagine it is not supported to take a base method from C# 11 which uses UnscopedRefAttribute, and override it in C# 10?

Reply:
`[UnscopedRef]` can only be applied to `struct` instance methods so overriding is not possible.

Reply:
a virtual method could have an out parameter like this, right?

Reply:
> A by-ref parameter on a virtual method could be impacted here, right?

Sorry, yes, virtual methods can have by-ref parameters that are overridden with/without `[UnscopedRef]`. We allow removing `[UnscopedRef]` in the override, and we report a warning or error when adding `[UnscopedRef]` in the override depending on `-langversion` - see `RefSafetyRulesAttribute_13` and `_14`.